compiler/qsc_eval/src/intrinsic/tests.rs

```diff
@@ -1173,3 +1173,75 @@ fn qubit_not_unique_three_qubit_error_second_third() {
         &expect!["qubits in gate invocation are not unique"],
     );
 }
+
+#[test]
+fn single_qubit_rotation_nan_error() {
+    check_intrinsic_output(
+        "",
+        indoc! {"{
+            use q = Qubit();
+            Rx(Microsoft.Quantum.Math.ArcSin(2.0), q);
```

Comment:
Should `ArcSin` do any parameter validation so it fails _before_ it returns NaN?

Reply:
I don't think so, because `NaN` is the defined result of `ArcSin` when the input value is outside of the accepted range (apparently that is common for trig libraries). A user used to be able to guard against these kinds of values in their own code with utility functions like `IsNaN` and `IsInfinite` (see the old math library [here](https://github.com/microsoft/qsharp-runtime/blob/3af588c549aa7d3d6d512e00e61ab697597b2fed/src/Simulation/QSharpFoundation/Math/FloatingPoint.qs#L46)), but we don't have equivalents for those currently. I'm not sure we actually need or want those functions back, I defer to @DmitryVasilevsky on that one.

Reply:
Makes sense to me. Thanks